py-polars/src/interop/arrow/to_py.rs

```diff
@@ -49,3 +55,39 @@ pub(crate) fn to_py_rb(
 
     Ok(record.to_object(py))
 }
+
+/// Export a series to a C stream via a PyCapsule according to the Arrow PyCapsule Interface
+/// https://arrow.apache.org/docs/dev/format/CDataInterface/PyCapsuleInterface.html
+pub(crate) fn series_to_stream<'py>(
+    series: &'py Series,
+    py: Python<'py>,
+) -> PyResult<Bound<'py, PyCapsule>> {
+    let field = series.field().to_arrow(CompatLevel::oldest());
+    let iter = Box::new(series.chunks().clone().into_iter().map(Ok)) as _;
+    let stream = ffi::export_iterator(iter, field);
+    let stream_capsule_name = CString::new("arrow_array_stream").unwrap();
+    PyCapsule::new_bound(py, stream, Some(stream_capsule_name))
+}
+
+pub(crate) fn dataframe_to_stream<'py>(
+    df: &'py DataFrame,
+    py: Python<'py>,
+) -> PyResult<Bound<'py, PyCapsule>> {
+    let schema_fields = df.schema().iter_fields().collect::<Vec<_>>();
+
+    let struct_field =
+        Field::new("", DataType::Struct(schema_fields)).to_arrow(CompatLevel::oldest());
+    let struct_data_type = struct_field.data_type().clone();
+
+    let iter = df
+        .iter_chunks(CompatLevel::oldest(), false)
+        .into_iter()
+        .map(|chunk| {
+            let arrays = chunk.into_arrays();
+            let x = arrow::array::StructArray::new(struct_data_type.clone(), arrays, None);
```

Comment:
Here I'm getting 
```
`struct_data_type` does not live long enough
```

I think this is easier in arrow-rs because the `RecordBatch` (`chunk` here) contains a reference to the schema. Whereas here in order to assemble the arrays in the record batch into a struct array we seem to need to know the data types from outside the closure? Or is there some other canonical way to construct a StructArray from a RecordBatch?